src/App.js

```diff
@@ -30,42 +33,98 @@ const App = () => {
   // This starts state off as a 2D array of JS objects with
   // empty value and unique ids.
   const [squares, setSquares] = useState(generateSquares());
+  const [currentPlayer, setCurrentPlayer] = useState(PLAYER_1);
+  const [winner, setWinner] = useState('');
+  const [turns, setTurns] = useState(0)
 
   // Wave 2
   // You will need to create a method to change the square 
   //   When it is clicked on.
   //   Then pass it into the squares as a callback
+  const onClickCallback = (id) => {
+    let row = 0
+    while (id > 2) {
+      id -= 3
+      row++
+    }
 
+    if (!squares[row][id]['value'] && !winner) {
+      const newSquares = [...squares]
+      newSquares[row][id]['value'] = currentPlayer
+      setSquares(newSquares)
 
-  const checkForWinner = () => {
+      const updatedPlayer = currentPlayer === PLAYER_1 ? PLAYER_2 : PLAYER_1
+      setCurrentPlayer(updatedPlayer)
+      const gameOver = checkForWinner(row, id)
+      if (!gameOver) {
+        if (turns === 8) {
+          setWinner('Tie')
```

Comment:
I would personally be more inclined to introduce an additional boolean piece of state specifically to to track whether the game was tied. If you _really_ prefer using the same winner state, then I'd suggest making a `const` to hold the special string indicating the tie condition. The same constant could then be used later when you're building the game result string.

Making a single const value helps protect us from typos in strings, which is always a risk with magic values.